src/main/java/org/terasology/managerInterface/nui/ManagementScreenLayer.java

```diff
@@ -0,0 +1,164 @@
+import org.terasology.spawning.OreonSpawnEvent;
+
+import java.util.*;
+
+/**
+ * The screen which is triggered when player interacts({@code e press}) with a Management book block.
+ */
+public class ManagementScreenLayer extends CoreScreenLayer {
+    private static final Logger logger = LoggerFactory.getLogger(ManagementScreenLayer.class);
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private LocalPlayer localPlayer;
+
+    @In
+    private PrefabManager prefabManager;
+
+    private UIList ownedVillagesList;
+    private UIList managedVillagesList;
+    private UILabel selectedVillageNameLabel;
+    private UILabel selectedVillageInfoLabel;
+
+    private List<VillageEntity> debugOwnedVillages;
+    private List<VillageEntity> debugManagedVillages;
+
+
+    @Override
+    public void initialise() {
+        ownedVillagesList = find(Constants.OWNED_VILLAGES_LIST_ID, UIList.class);
+        managedVillagesList = find(Constants.MANAGED_VILLAGES_LIST_ID, UIList.class);
+        selectedVillageNameLabel = find(Constants.SELECTED_VILLAGE_NAME_LABEL_ID, UILabel.class);
+        selectedVillageInfoLabel = find(Constants.SELECTED_VILLAGE_INFO_LABEL_ID, UILabel.class);
+
+        List<String> cityNames = new ArrayList<String>();
+        cityNames.add("Lordaeron");
+        cityNames.add("Prague");
+        cityNames.add("Kategad");
+        cityNames.add("London");
+        cityNames.add("Berlin");
+        cityNames.add("Helsinki");
+        //
+        cityNames.add("Viena");
+        cityNames.add("NewYork");
+        cityNames.add("Brno");
+        cityNames.add("Katowice");
+
+        debugOwnedVillages = new ArrayList<VillageEntity>();
+        for (int i = 0; i < 6; i++) {
+            VillageEntity city = new VillageEntity(cityNames.get(i),"nightmaredev");
+            debugOwnedVillages.add(city);
+        }
+
+        debugManagedVillages = new ArrayList<VillageEntity>();
+        for (int i = 6; i < 10; i++) {
+            VillageEntity city = new VillageEntity(cityNames.get(i),"xXxSlayer1337");
+            debugManagedVillages.add(city);
+        }
+
+        ownedVillagesList.subscribeSelection((widget, item) -> {
+            DisplayVillageInfo((VillageEntity)item);
+        });
+        managedVillagesList.subscribeSelection((widget, item) -> {
+            DisplayVillageInfo((VillageEntity)item);
+        });
+        populateLists();
+    }
+
+    /**
+     * Popultes the text label fields in the screen with the items required for spawning.
+     * @param cityName The Oreon prefab for which the label is being set.
+     * @param label The label to be set.
+     */
+    private void populateUiLabels(String cityName, UILabel label) {
+        StringBuilder text = new StringBuilder("Teleport to " + cityName);
+        label.setText(text.toString());
+    }
+
+    private void writeOutCity(String cityName) {
+        logger.info("Teleported player to " + cityName);
```

Comment:
Seeing this in its own method puzzles me a bit?

Reply:
Oh. This is piece of forgotten code I used to just get the cityname. I will remove this one.